### PR TITLE
chore(core): make DROP statement compatible with ENT

### PR DESCRIPTION
**Pull request by marregui**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 4 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|2|4|50.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 4 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|2|4|50.00%|

File: core/src/main/java/io/questdb/griffin/SqlCompiler.java [hunk 1]
```diff
@@ -3280,22 +3280,21 @@ private CompiledQuery executorSelector(SqlExecutionContext executionContext) thr
                     if (tok == null || Chars.equals(tok, ';')) {
                         return dropTable(executionContext, tableName, tableNamePosition, hasIfExists);
                     }
-                    throw parseErrorUnexpected("[;]", tok);
+                    return unknownDropTableSuffix(executionContext, tok, tableName, tableNamePosition, hasIfExists);
                 }
-
                 // DROP ALL TABLES [;]
-                if (SqlKeywords.isAllKeyword(tok)) {
+                else if (SqlKeywords.isAllKeyword(tok)) {
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/SqlCompiler.java:3286]: nit: `else` is redundant here
```suggestion
                if (SqlKeywords.isAllKeyword(tok)) {
```

File: core/src/test/java/io/questdb/test/cutlass/pgwire/PGJobContextTest.java [hunk 2]
```diff
@@ -2381,16 +2381,16 @@ public void testDropTable() throws Exception {
         String[][] sqlExpectedErrMsg = {
                 {"drop table doesnt", "ERROR: table does not exist [table=doesnt]"},
                 {"drop table", "ERROR: expected IF EXISTS table-name"},
-                {"drop doesnt", "ERROR: expected TABLE table-name or ALL TABLES, found unexpected [token='doesnt']"},
-                {"drop", "ERROR: expected TABLE"},
+                {"drop doesnt", "ERROR: 'table' expected"},
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/cutlass/pgwire/PGJobContextTest.java:2384]: This error doesn't seem to be the correct one as now we support DROP TABLE and DROP ALL TABLES. It should be enough to update the messages in the `unknownDropStatement` and `unknownDropTableSuffix` methods to keep the tests as they are.

File: core/src/test/java/io/questdb/test/griffin/DropStatementTest.java [hunk 3]
```diff
@@ -41,8 +41,7 @@ public class DropStatementTest extends AbstractGriffinTest {
     /* **
      * DROP can be followed by:
      * - TABLE name [;]
-     * - TABLES name(,name)* [;]
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/griffin/DropStatementTest.java:44]: nit: a question around #3438. We no longer support DROP TABLE [table, ...] syntax, right? AFAIR it used to be supported, yet not documented properly.

File: core/src/test/java/io/questdb/test/cutlass/pgwire/PGJobContextTest.java (hunk 2 above)
Comment (marregui) [reply core/src/test/java/io/questdb/test/cutlass/pgwire/PGJobContextTest.java]: should have thought that!, done

File: core/src/test/java/io/questdb/test/griffin/DropStatementTest.java (hunk 3 above)
Comment (marregui) [reply core/src/test/java/io/questdb/test/griffin/DropStatementTest.java:44]: nope, I can add on demand in a follow up pr.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 5 (60.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|3|5|60.00%|

**Review by puzpuzpuz — APPROVED** [review]

**Pull request merged**